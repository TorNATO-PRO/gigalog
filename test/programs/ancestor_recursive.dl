parent(john, mary).
parent(mary, bob).
parent(bob, alice).

ancestor(X, Y) :- parent(X, Y).
ancestor(X, Y) :- parent(X, Z), ancestor(Z, Y).
