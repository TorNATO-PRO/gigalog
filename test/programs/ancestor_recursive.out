ancestor(bob,alice)
ancestor(john,alice)
ancestor(john,bob)
ancestor(john,mary)
ancestor(mary,alice)
ancestor(mary,bob)
parent(bob,alice)
parent(john,mary)
parent(mary,bob)
